### PR TITLE
(minor) Prevent default language file (en.json) from being fetched twice when loading the app

### PR DESCRIPTION
**Pull request by amirtoole**

### CODE REVIEW

File: src/client/app/frameworks/i18n.framework/services/multilingual.service.ts [hunk 1]
```diff
@@ -43,39 +43,42 @@ export const multilingualReducer: ActionReducer<MultilingualStateI> = (state: Mu
...
+
   // default supported languages
   // see main.ts bootstrap for example of how to provide different value
   public static SUPPORTED_LANGUAGES: Array<ILang> = [
-    { code: 'en', title: 'English' }
+    {code: 'en', title: 'English'}
   ];
-  
+
+  private static DEFAULT_LANG: string = 'en';
```
Comment (NathanWalker) [inline src/client/app/frameworks/i18n.framework/services/multilingual.service.ts:53]: Funny to see a `private` static variable?
Would be nice to set this during bootstrap for any platform so maybe make `public`

**Comment by NathanWalker** [comment]
Thanks for this 👍  Just 1 comment.

**Comment by amirtoole** [comment]
Yup; I agree. Will push that now.

On a similar topic: what do you think about persisting the language selection using local storage?
Ideally, I'd like to have a more robust system that checks the following to determine start language (first hit wins):
1. URL param
2. Local storage
3. browser/platform
4. Fallback to default

Local storage would be set if user manually changes language or if URL param exists. If you agree I'll PR that separately.

**Comment by NathanWalker** [comment]
Thanks @hellofornow yes that would be nice. I have had plans to integrate some sort of low level storage option into core.framework, it would be nice if it used https://github.com/mozilla/localForage and I could write a NativeScript service which could `provide` on top of that to write to native storage options.
If you wanted to try a separate PR for something like this, I'm open to it, would need unit tests etc.

You have a minor error that is failing the build:
/home/travis/build/NathanWalker/angular2-seed-advanced/src/client/app/frameworks/i18n.framework/services/multilingual.service.ts
      line 72  col 18  != should be !==

**Comment by amirtoole** [comment]
I'll add localForage and get that working; definitely need persistence for my work project.

I'm not sure this PR is that useful; in fact it's not working. If I override DEFAULT_LANGUAGE to 'fr' and keep my browser locale to 'en', it fetches en.json twice and doesn't fetch 'fr.json'. Removing a string from 'en.json' causes runtime exception. Better to close this PR and I'll fix the duplicate download at the same time as adding persistance.

I did find a couple other places that hardcode 'en' that will need updating:
(painful) https://github.com/NathanWalker/angular2-seed-advanced/blob/master/src/client/app/frameworks/i18n.framework/services/multilingual.service.ts#L24
(trivial) https://github.com/NathanWalker/angular2-seed-advanced/blob/master/src/client/app/frameworks/i18n.framework/components/lang-switcher.component.ts#L37

**Comment by NathanWalker** [comment]
Yep let's close this. If working on a new PR with persistence in mind, I would just set `initialState` here to nothing:
https://github.com/NathanWalker/angular2-seed-advanced/blob/master/src/client/app/frameworks/i18n.framework/services/multilingual.service.ts#L24
instead something like this:

```
export interface MultilingualStateI {
  lang?: string;
}

export const multilingualReducer: ActionReducer<MultilingualStateI> = (state: MultilingualStateI = {}, action: Action) => {
```

And then grab the language from persistence if set otherwise fallback to static variable.

**Pull request closed without merging**